src/Http/Wolverine.Http.Marten/CompiledQueryWriterPolicy.cs

```diff
@@ -14,82 +15,65 @@ public class CompiledQueryWriterPolicy : IResourceWriterPolicy
+                        "Cannot find correct method for Marten.AspNetCore.QueryableExtensions.WriteArray to bind to. Incompatible Versions?"));
+                }
 
-    public override void GenerateCode(GeneratedMethod method, ISourceWriter writer)
-    {
-        writer.WriteComment("Run the compiled query and stream the response");
-        writer.Write($"await Marten.AspNetCore.QueryableExtensions.WriteOne({_documentSession?.Usage}, {_compiledQuery.Usage}, {_httpContext?.Usage});");
-        Next?.GenerateCode(method, writer);
-    }
-    public override IEnumerable<Variable> FindVariables(IMethodVariables chain)
-    {
-        _documentSession = chain.FindVariable(typeof(IDocumentSession));
-        yield return _documentSession;
-        _httpContext = chain.FindVariable(typeof(HttpContext));
-        yield return _httpContext;
-        foreach (var variable in base.FindVariables(chain)) yield return variable;
-    }
-}
+                methodToCall = method[0].MakeGenericMethod(arguments);
+            }
+            else
+            {
+                var method = typeof(QueryableExtensions).GetMethod(nameof(QueryableExtensions.WriteOne));
+                if (method is null)
+                {
+                    throw new CodeGenerationException(typeof(QueryableExtensions), new Exception(
+                        "Cannot find correct method for Marten.AspNetCore.QueryableExtensions.WriteOne to bind to. Incompatible Versions?"));
+                }
 
-public class MartenWriteArrayCodeFrame : AsyncFrame
-{
-    private Variable? _documentSession;
-    private Variable? _httpContext;
-    private readonly Variable _compiledQuery;
+                methodToCall = method.MakeGenericMethod(arguments);
+            }
 
-    public MartenWriteArrayCodeFrame(Variable variable)
-    {
-        _compiledQuery = variable;
-    }
 
-    public override void GenerateCode(GeneratedMethod method, ISourceWriter writer)
-    {
-        writer.WriteComment("Run the compiled query and stream the response");
-        writer.Write($"await Marten.AspNetCore.QueryableExtensions.WriteArray({_documentSession?.Usage}, {_compiledQuery.Usage}, {_httpContext?.Usage});");
-        Next?.GenerateCode(method, writer);
-    }
-    public override IEnumerable<Variable> FindVariables(IMethodVariables chain)
-    {
-        _documentSession = chain.FindVariable(typeof(IDocumentSession));
-        yield return _documentSession;
-        _httpContext = chain.FindVariable(typeof(HttpContext));
-        yield return _httpContext;
-        foreach (var variable in base.FindVariables(chain)) yield return variable;
+            var call = new MethodCall(typeof(QueryableExtensions), methodToCall)
+                            {
+                                Arguments =
+                                {
+                                    [1] = result,
+                                    [3] = new StaticVariable(typeof(string), "\"application/json\""),
```

Comment:
There's also a `Constant` for this kind of literal value like so: `Constant.ForString("application/json")`, `Constant.For(200)`